src/libraries/System.Private.Xml/src/System/Xml/Serialization/SoapAttributes.cs

```diff
@@ -20,11 +21,11 @@ internal enum SoapAttributeFlags
     public class SoapAttributes
     {
         private bool _soapIgnore;
-        private SoapTypeAttribute _soapType;
-        private SoapElementAttribute _soapElement;
-        private SoapAttributeAttribute _soapAttribute;
-        private SoapEnumAttribute _soapEnum;
-        private object _soapDefaultValue;
+        private SoapTypeAttribute? _soapType;
+        private SoapElementAttribute? _soapElement;
+        private SoapAttributeAttribute? _soapAttribute;
+        private SoapEnumAttribute? _soapEnum;
+        private object? _soapDefaultValue;
 
         public SoapAttributes()
         {
```

Comment:
But `provider` cannot be `null` there right (since it's declared as non-nullable)?

Reply:
Aren't we supposed to avoid `NullReferenceExceptions`? I think we are missing an actual nullcheck in line 35 - We need to throw `ArgumentNullException(nameof(provider))`, before calling `provider.GetCustomAttributes()`.

Reply:
I've updated https://github.com/dotnet/runtime/issues/41061 with this case since this won't meet bar for 5.0